src/IncludedCommands/Resources/PersistentBans.luau

```diff
@@ -149,20 +152,128 @@ end
+end
+
+
+--[[
+Unbans user ids
+--]]
+function PersistentBans.UnbanIds(self: PersistentBans, Config: UnbanConfigType, ModeratorId: number?, ModeratorName: string?): (boolean, string?)
+    self:Log(Config, ModeratorId, ModeratorName)
+    return pcall(function()
+        Players:UnbanAsync(Config)
+    end)
 end
 
+--[[
+Logs an action
+--]]
+function PersistentBans.Log(self: PersistentBans, Config: BanConfigType | UnbanConfigType, ModeratorId: number?, ModeratorName : string?)
+    if self.LogsDataStore then
+        task.spawn(function()
+            if (Config :: BanConfigType).DisplayReason then
+                self.LogsDataStore:Update("PersistentBanActionsLog", function(OldLogs)
+                    OldLogs = OldLogs or {}
+                    table.insert(OldLogs, {
+                        Time = os.time(),
+                        Id = ModeratorId or 0,
+                        Name = ModeratorName or "NexusAdmin",
+                        Action = 0,
+                        EffectedUserIds = Config.UserIds,
+                        Duration = math.max(-1, Config.Duration),
+                        ExperienceWide = Config.ApplyToUniverse,
+                        IncludeAlts = not Config.ExcludeAltAccounts
+                    })
+                    return OldLogs
+                end)
+            else
+                self.LogsDataStore:Update("PersistentBanActionsLog", function(OldLogs)
+                    OldLogs = OldLogs
+                    table.insert(OldLogs, {
+                        Time = os.time(),
+                        Id = ModeratorId or 0,
+                        Name = ModeratorName or "NexusAdmin",
+                        Action = 1,
+                        EffectedUserIds = Config.UserIds,
+                        ExperienceWide = Config.ApplyToUniverse,
+                    })
+                    return OldLogs
+                end)
+            end
+        end)
+    end
+end
 
+--[[
+Returns all persistent ban logs
+--]]
+function PersistentBans.GetLogs(self: PersistentBans)
+    if self.LogsDataStore then
+        return self.LogsDataStore:Get("PersistentBanActionsLog") or {}
+    else
+        return {"Persistent Ban Logs failed to initialize."}
+    end
+end
 
+--[[
+Try and filter out Persistent Ban Immune UserIds of both in server players and out of server players.
+--]]
+function PersistentBans.FilterIds(self: PersistentBans, Ids: {number}, ModeratorId: number): ({number}, {number})
+    local Filtered = {}
+    local Immune = {}
+    local PersistentBanImmuneLevel = self.Api.Configuration.CommandConfigurations.PersistentBanImmuneLevel
+    local AdminLevel = PersistentBanImmuneLevel
+    if ModeratorId > 0 then
+        AdminLevel = self.Api.Authorization:GetAdminLevel(Players:GetPlayerByUserId(ModeratorId))
+    end
+    for i, Id in Ids do
+        if self.Api.Authorization:EvaluateAdminLevelAsync(Id) >= math.min(PersistentBanImmuneLevel, AdminLevel) then
```

Comment:
`EvaluateAdminLevelAsync` is not exposed in the API. I *think* it is probably safe to add it to the types + documentation.